slang-checks/src/main/java/com/sonarsource/slang/checks/BadFunctionNameCheck.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * SonarSource SLang
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonarsource.slang.checks;
+
+import com.sonarsource.slang.api.FunctionDeclarationTree;
+import com.sonarsource.slang.checks.api.InitContext;
+import com.sonarsource.slang.checks.api.SlangCheck;
+import java.util.regex.Pattern;
+import org.sonar.check.Rule;
+import org.sonar.check.RuleProperty;
+
+@Rule(key = "S100")
+public class BadFunctionNameCheck implements SlangCheck {
+
+  private static final String DEFAULT_FORMAT = "^[a-z][a-zA-Z0-9]*$";
+
+  @RuleProperty(
+    key = "format",
+    description = "Regular expression used to check the function names against.",
+    defaultValue = DEFAULT_FORMAT
+  )
+  public String format = DEFAULT_FORMAT;
+
+  private Pattern pattern = null;
+
+  private String message(String name) {
+    return "Rename function " + name + " to match the regular expression " + format;
+  }
+
+  @Override
+  public void initialize(InitContext init) {
+    pattern = Pattern.compile(format);
+    init.register(FunctionDeclarationTree.class, (ctx, fnDeclarationTree) -> {
+      String name = fnDeclarationTree.name().name();
+      if (!pattern.matcher(name).matches()) {
```

Comment:
The function name may be null, at least in Kotlin... Unfortunately, it seems we're not able to test that in slang textual representation...